CMakeLists.txt

```diff
@@ -9,6 +9,23 @@ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/")
 # Should clang-tidy errors be automatically fixed?
 option(FIX_CLANG_TIDY_ERRORS OFF)
 
+# Enable coverage reports
+option(ENABLE_COVERAGE OFF)
+if(ENABLE_COVERAGE)
+    include(CodeCoverage)
+    append_coverage_compiler_flags()
+
+    setup_target_for_coverage_lcov(
+        NAME coverage-core
+        EXECUTABLE core/tests/cubos-core-tests
+        DEPENDENCIES cubos-core-tests
+
+        EXCLUDE "core/lib/*" "core/tests/*"
+
+        LCOV_ARGS "--no-external"
+    )
+endif()
```

Comment:
Shouldn't this be moved to the `core` CMakeLists.txt?

Reply:
in the future I also want to expand this to the engine, whenever we start unit testing it as well.
it has already been renamed to "coverage" and the executable could be changed to `core/tests/cubos-core-tests && engine/tests/cubos-engine-tests`